.github/workflows/kani.yml

```
@@ -226,3 +226,61 @@ jobs:
           if-no-files-found: error
           # Aggressively short retention: we don't really need these
           retention-days: 3
+
+  perf-benchcomp:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Save push event HEAD and HEAD~ to environment variables
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          echo "NEW_REF=${{ github.event.after}}" | tee -a "$GITHUB_ENV"
+          # We want to compare with $NEW_REF~. But we can't know what
+          # that ref actually is until we clone the repository, so for
+          # now make it equal to $NEW_REF
+          echo "OLD_REF=${{ github.event.after }}" | tee -a "$GITHUB_ENV"
```

what about making this something like:
```suggestion
          echo "OLD_REF=${{ github.event.after }}~" | tee -a "$GITHUB_ENV"
```
and removing the extra checkout step?

---

If this doesn't work, another possibility would be:
```suggestion
          echo "OLD_REF=HEAD~" | tee -a "$GITHUB_ENV"
```

---

These don't work because the 'checkout' action needs to send the actual ref to the server in order to do a shallow clone. We can't specify things like `HEAD~` or `abc123~` as an argument to the checkout@v3 action. So I'm instead specifying HEAD, and cloning to a depth of 2 rather than 1 so that I can subsequently checkout HEAD~.